### PR TITLE
Feedback

### PR DESCRIPTION
**Pull request by aturaab40**
:wave:! GitHub Classroom created this pull request as a place for your teacher to leave feedback on your work. It will update automatically. **Don’t close or merge this pull request**, unless you’re instructed to do so by your teacher.

In this pull request, your teacher can leave comments and feedback on your code. Click the **Subscribe** button to be notified if that happens.

Click the **Files changed** or **Commits** tab to see all of the changes pushed to `master` since the assignment started. Your teacher can see this too.

<details>
<summary>
<strong>Notes for teachers</strong>
</summary>

Use this PR to leave feedback. Here are some tips:

- Click the **Files changed** tab to see all of the changes pushed to `master` since the assignment started. To leave comments on specific lines of code, put your cursor over a line of code and click the blue **+** (plus sign). To learn more about comments, read “[Commenting on a pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request)”.
- Click the **Commits** tab to see the commits pushed to `master`. Click a commit to see specific changes.
- If you turned on autograding, then click the **Checks** tab to see the results.
- This page is an overview. It shows commits, line comments, and general comments. You can leave a general comment below.

For more information about this pull request, read “[Leaving assignment feedback in GitHub](https://classroom.github.com/help/leaving-feedback-in-github)”.
</details>

Subscribed: @touraye

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const students = [
+let students = [
```
Comment (aturaab40) [inline main.js:1]: why do you change this from const to let?

File: main.js [hunk 2]
```diff
@@ -50,4 +50,79 @@ const students = [
...
+    skill : 6,
+    interest :[ "programming", "research", "Cybersecurity", "Networking", "computer repair", "social Engineer", "workout", "riding"]  
+}
+
+// add the new entry(newStudents) to the students Array (q-1)
+students.push(newStudent);
+// console.log(students);
+
+// print all the names of the students (q-2)
+for (let value of Object.values(students)) {
```
Comment (aturaab40) [inline main.js:67]: you can simply say:
`for (let value of students) {

}`

File: main.js [hunk 3]
```diff
@@ -50,4 +50,79 @@ const students = [
...
+    let listOfStud = [];
+     listOfStud = value.name;
+    // console.log(listOfStud);
+}
+
+
+    // the interest of all the studetns from jam city (q-3)
+ let student = students.filter(item => {
```
Comment (aturaab40) [inline main.js:75]: If you use filter there is no need to use the condition inside the filter function

File: main.js [hunk 4]
```diff
@@ -50,4 +50,79 @@ const students = [
...
+            // console.log(set.interest);
+        }
+        return student;
+    });
+    // console.log(studInterest);
+   
+        
+
+
+// count how many students have skill set more than 5 (q-4)
```
Comment (aturaab40) [inline main.js:92]: Same thing here

File: main.js (hunk 1 above)
Comment (touraye) [reply main.js:1]: Sorry I thought I replace it the way it was. I was trying to reassign the students array.

**Comment by touraye** [comment]
Thanks for reviews. Noted!

**Comment by aturaab40** [comment]
Update your solution

**Pull request merged**